### PR TITLE
Fix issue 15576 - Skip over the parent expression when parent is NSpace.

### PR DESCRIPTION
**Pull request by LemonBoy**
Don't have a windows machine handy so I'll just abuse the CI.
Long story short, the protection attributes should be calculated by taking into account that the enclosing parent symbol might be a `NScope` such as in
```D
module d;
extern (C++, op) extern __gshared int variable;
```
I think that something should be done for declarations inside mixins and/or templates (as a bonus, trying to enclose the declaration above in a mixin results in a `undefined identifier: 'variable'`), not sure about that.

### CODE REVIEW

**Comment by kinke** [comment]
https://github.com/dlang/dmd/pull/6233 ;)

**Comment by kinke** [comment]
I assumed C++ namespaces could only be nested inside other C++ namespaces or directly reside inside the module (but haven't checked it), so I restricted it to a single check.

**Comment by WalterBright** [comment]
Needs test case(s).

**Comment by LemonBoy** [comment]
Let's play rock paper scissor to find our what PR's gonna get merged heh

I've tried to stay as close as the original code as possible and leave the `isModule` check in place.

I've ripped off the tests from @kinke's PR and modified them to check the case of nested namespace declaration too.

File: src/cppmangle.d [hunk 1]
```diff
@@ -1467,7 +1467,13 @@ else static if (TARGET_WINDOS)
             buf.writeByte('?');
             mangleIdent(d);
             assert((d.storage_class & STCfield) || !d.needThis());
-            if (d.parent && d.parent.isModule()) // static member
+            Dsymbol parent = d.toParent();
+            // skip the enclosing 'export' scopes
```
Comment (kinke) [inline src/cppmangle.d]: `'export' scopes` = C++ namespaces. I don't think the comment is needed though, the code is self-explaining IMO.

**Pull request merged**